capstone/src/main/java/com/google/sps/servlets/GroupPostDataServlet.java

```diff
@@ -36,27 +36,26 @@
 import com.google.sps.Objects.Post;
 import com.google.sps.Objects.Comment;
 import com.google.sps.Objects.response.PostResponse;
+import error.ErrorHandler;
 
 @WebServlet("/group-post")
 public class GroupPostDataServlet extends HttpServlet {
 
+  private ErrorHandler errorHandler = new ErrorHandler();
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
-    String userId = "";
-    UserService userService = UserServiceFactory.getUserService();
-    if (userService.isUserLoggedIn()) {
-      userId = userService.getCurrentUser().getUserId();
-    }
-
+    String userId = this.getUserId(response);
+    if (userId.equals("")) return;
```

Comment:
In case this is not part of the previous PR, you can also use the AuthenticatedServlet here to get the userID instead

Reply:
changed in previous PR